root/edit/details/AddArtist.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {
+  artistBeginAreaLabel,
+  artistBeginLabel,
+  artistEndAreaLabel,
+  artistEndLabel,
+} from '../../artist/utils';
+import {addColon, l} from '../../static/scripts/common/i18n';
+import {lp_attributes} from '../../static/scripts/common/i18n/attributes';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import isDateEmpty from '../../static/scripts/common/utility/isDateEmpty';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+import formatIsni from '../../utility/formatIsni';
+
+type AddArtistEditT = {|
+  ...EditT,
+  +display_data: {|
+    ...CommentRoleT,
+    ...DatePeriodRoleT,
+    +area: AreaT | null,
+    +artist: AreaT,
+    +begin_area: AreaT | null,
+    +end_area: AreaT | null,
+    +gender: GenderT | null,
+    +ipi_codes: $ReadOnlyArray<string>,
+    +isni_codes: $ReadOnlyArray<string>,
```

Comment:
`ipi_codes` and `isni_codes` can both be `null` (old edits currently break due to the `.length` checks below).